src/players/VAST.js

```diff
@@ -0,0 +1,211 @@
+    this.container.play()
+  }
+
+  pause () {
+    this.container.pause()
+  }
+
+  stop () {
+    this.container.stop()
+  }
+
+  // only allow rewind
+  seekTo (seconds) {
+    if (seconds < this.container.getCurrentTime()) {
+      this.container.seekTo(seconds)
+    }
+  }
+
+  setVolume (fraction) {
+    this.container.setVolume(fraction)
+  }
+
+  mute = () => {
+    this.container.mute()
+  };
+
+  unmute = () => {
+    this.container.unmute()
+  };
+
+  getDuration () {
+    return this.container.getDuration()
+  }
+
+  getCurrentTime () {
+    return this.container.getCurrentTime()
+  }
+
+  getSecondsLoaded () {
+    return this.container.getSecondsLoaded()
+  }
+
+  ref = (container) => {
+    this.container = container
+  };
+
+  onAdClick = () => {
+    const {state: {tracker}} = this
+    tracker.click()
+  }
+
+  openAdLink (url) {
+    window.open(url, '_blank')
+  }
+
+  // track ended
+  onEnded = (event) => {
+    const {props: {onEnded}, state: {tracker}} = this
+    tracker.complete()
+    onEnded(event)
+  }
+
+  // track error
+  onError = (event) => {
+    const {props: {onError}, state: {tracker}} = this
+    vast.util.track(tracker.ad.errorURLTemplates, { ERRORCODE: 405 })
+    onError(event)
+  }
+
+  // track pause
+  onPause = (event) => {
+    const {props: {onPause}, state: {tracker}} = this
+    tracker.setPaused(true)
+    onPause(event)
+  }
+
+  // track play
+  onPlay = (event) => {
+    const {props: {onPlay}, state: {tracker}} = this
+    tracker.setPaused(false)
+    onPlay(event)
+  }
+
+  onProgress = (event) => {
+    const {props: {onProgress}, state: {tracker}} = this
+    tracker.setProgress(event.playedSeconds)
+    onProgress(event)
+  }
+
+  // track load and duration
+  onReady = (event) => {
+    const {props: {onReady}, state: {tracker}} = this
+    tracker.load()
+    if (Number.isNaN(tracker.assetDuration)) {
+      tracker.assetDuration = this.container.getDuration()
+    }
+    onReady(event)
+  }
+
+  // track volume change
+  onVolumeChange = (event) => {
+    const {props: {onVolumeChange}, state: {tracker}} = this
+    tracker.setMuted(this.container.muted)
+    onVolumeChange(event)
+  }
+
+  render () {
+    const {sources, tracker: clickTrackingURLTemplate} = this.state
+    const { width, height } = this.props
+    const wrapperStyle = {
+      cursor: clickTrackingURLTemplate ? 'pointer' : 'default'
+    }
+    const videoStyle = {
+      width: width === 'auto' ? width : '100%',
+      height: height === 'auto' ? height : '100%'
+    }
+    if (!sources.length) return null
+
+    return (
+      <div onClick={this.onAdClick} style={wrapperStyle}>
+        <FilePlayer
+          {...this.props}
+          onEnded={this.onEnded}
+          onError={this.onError}
+          onPause={this.onPause}
+          onPlay={this.onPlay}
+          onProgress={this.onProgress}
+          onReady={this.onReady}
+          onVolumeChange={this.onVolumeChange}
+          ref={this.ref}
```

Comment:
This should be changed to createRef:
```javascript
...
container = React.createRef();
...
```
`ref={this.container}`

Reply:
not changing this